src/collapse/ghcprof.rs

```diff
@@ -0,0 +1,222 @@
+            if line.is_empty() {
+                break;
+            } else {
+                self.on_line(line, &mut occurrences, &cols)?;
+            }
+        }
+
+        // Write the results...
+        occurrences.write_and_clear(writer)?;
+
+        // Reset the state...
+        self.current_cost = 0;
+        self.stack.clear();
+        Ok(())
+    }
+
+    /// Check for start line of a call graph.
+    fn is_applicable(&mut self, input: &str) -> Option<bool> {
+        let mut input = input.as_bytes();
+        let mut line = String::new();
+        loop {
+            line.clear();
+            if let Ok(n) = input.read_line(&mut line) {
+                if n == 0 {
+                    break;
+                }
+            } else {
+                return Some(false);
+            }
+
+            if line
+                .split_whitespace()
+                .take(START_LINE.len())
+                .eq(START_LINE.iter().cloned())
+            {
+                return Some(true);
+            }
+        }
+        None
+    }
+}
+
+impl From<Options> for Folder {
+    fn from(opt: Options) -> Self {
+        Folder {
+            opt,
+            ..Default::default()
+        }
+    }
+}
+
+impl Folder {
+    // Handle call graph lines of the form:
+    //
+    // MAIN           MAIN ...
+    //  CAF           Options.Applicative.Builder ...
+    //   defaultPrefs Options.Applicative.Builder ...
+    //    idm         Options.Applicative.Builder ...
+    //    prefs       Options.Applicative.Builder ...
+    //   fullDesc     Options.Applicative.Builder ...
+    //   hidden       Options.Applicative.Builder ...
+    //   option       Options.Applicative.Builder ...
+    //    metavar     Options.Applicative.Builder ...
+    //  CAF           Options.Applicative.Builder.Internal ...
+    //   internal     Options.Applicative.Builder.Internal ...
+    //   noGlobal     Options.Applicative.Builder.Internal ...
+    //   optionMod    Options.Applicative.Builder.Internal ...
+
+    fn on_line(
+        &mut self,
+        line: &str,
+        occurrences: &mut Occurrences,
+        cols: &Cols,
+    ) -> io::Result<()> {
+        if let Some(indent_chars) = line.find(|c| c != ' ') {
+            let prev_len = self.stack.len();
+            let depth = indent_chars;
+
+            if depth < prev_len {
+                // If the line is not a child, pop stack to the stack before the new depth
+                self.stack.truncate(depth);
+            } else if depth != prev_len {
+                return invalid_data_error!("Skipped indentation level at line:\n{}", line);
+            }
+            // There can be non-ascii names so take care to char offset not byte offset
+            let string_range = |col_start: usize| {
+                line.chars()
+                    .skip(col_start)
+                    .skip_while(|c| c.is_whitespace())
+                    .take_while(|c| !c.is_whitespace())
```

Comment:
This won't work for fields that have spaces _in_ them. To do that you'd need to also store the _end_ index of a column, read to the end, and then `trim_right`. Probably not worth it, but worth flagging in a comment in case someone every thinks this _might_ be appropriate for fields with "spaced" values.

Reply:
added a comment to say the expected targets do not contain whitespace